src/openloco/windows/constructionwnd.cpp

```diff
@@ -1,75 +1,4836 @@
+        for (uint8_t i = 0; i < objectmgr::get_max_objects(object_type::airport); i++)
+        {
+            auto airportObj = objectmgr::get<airport_object>(i);
+            if (airportObj == nullptr)
+                continue;
+            if (currentYear < airportObj->designed_year)
+                continue;
+            if (currentYear > airportObj->obsolete_year)
+                continue;
+            _stationList[airportCount] = i;
+            airportCount++;
+        }
+        _stationList[airportCount] = 0xFF;
+
+        sortList(_stationList, std::size(_stationList));
+    }
+
+    // 0x0048D753
+    static void refreshDockList()
+    {
+        auto currentYear = current_year();
+        auto dockCount = 0;
+        for (uint8_t i = 0; i < objectmgr::get_max_objects(object_type::dock); i++)
+        {
+            auto dockObj = objectmgr::get<dock_object>(i);
+            if (dockObj == nullptr)
+                continue;
+            if (currentYear < dockObj->designed_year)
+                continue;
+            if (currentYear > dockObj->obsolete_year)
+                continue;
+            _stationList[dockCount] = i;
+            dockCount++;
+        }
+        _stationList[dockCount] = 0xFF;
+
+        sortList(_stationList, std::size(_stationList));
+    }
+
+    // 0x0048D678
+    static void refreshRoadStationList()
+    {
+        auto trackType = _trackType & ~(1 << 7);
+        auto stationListCopy = _stationList;
+        auto currentYear = current_year();
+        auto roadObj = objectmgr::get<road_object>(trackType);
+        auto roadStationCount = 0;
+
+        for (auto i = 0; i < roadObj->num_stations; i++)
+        {
+            auto station = roadObj->stations[i];
+            if (station == 0xFF)
+                continue;
+            auto roadStationObj = objectmgr::get<road_station_object>(station);
+            if (currentYear < roadStationObj->designed_year)
+                continue;
+            if (currentYear > roadStationObj->obsolete_year)
+                continue;
+            _stationList[roadStationCount] = station;
+            roadStationCount++;
+        }
+
+        for (uint8_t i = 0; i < objectmgr::get_max_objects(object_type::road_station); i++)
+        {
+            auto roadStationObj = objectmgr::get<road_station_object>(i);
+            if (roadStationObj == nullptr)
+                continue;
+            for (auto j = 0; j < roadStationObj->var_20; j++)
+            {
+                if (trackType != roadStationObj->num_compatible)
+                    continue;
+                if (currentYear < roadStationObj->designed_year)
+                    continue;
+                if (currentYear > roadStationObj->obsolete_year)
+                    continue;
+                for (size_t k = 0; k < std::size(_stationList); k++)
+                {
+                    auto station = stationListCopy[k];
+                    if (stationListCopy == _stationList)
```

Comment:
This isn't correct it is checking if `&_stationList[k] == &_stationList[roadStationCount]`